hedera-node/src/main/java/com/hedera/services/store/AccountStore.java

```diff
@@ -81,6 +82,38 @@ public Account loadAccount(Id id) {
 		return account;
 	}
 
+	/**
+	 * Wraps {@link AccountStore#loadAccount(Id)} with a try catch block
+	 * and throws the given code if an exception occurs due to an invalid account.
+	 *
+	 * <b>IMPORTANT:</b> Changes to the returned model are not automatically persisted
+	 * to state! The altered model must be passed to {@link AccountStore#persistAccount(Account)}
+	 * in order for its changes to be applied to the Swirlds state, and included in the
+	 * {@link com.hedera.services.state.submerkle.ExpirableTxnRecord} for the active transaction.
+	 *
+	 * @param id the account to load
+	 * @param code the {@link ResponseCodeEnum} to fail with if the account is deleted/missing
+	 * @return a usable model of the account if available
+	 */
+	public Account loadAccountOrFailWith(Id id, ResponseCodeEnum code) {
+		Account account;
+		try {
+			final var key = new MerkleEntityId(id.getShard(), id.getRealm(), id.getNum());
+			final var merkleAccount = accounts.get().get(key);
+
+			validateUsable(merkleAccount);
+
+			account = new Account(id);
+			account.setExpiry(merkleAccount.getExpiry());
+			account.initBalance(merkleAccount.getBalance());
+			account.setAssociatedTokens(merkleAccount.tokens().getIds().copy());
+			account.setOwnedNfts(merkleAccount.getNftsOwned());
+		} catch (InvalidTransactionException ite) {
```

Comment:
There is a discrepancy in the behaviour. In the previous logic, if `ACCOUNT_EXPIRED_AND_PENDING_REMOVAL` was thrown, by the validation logic, we were not throwing the `INVALID_TREASURY_ACCOUNT_FOR_TOKEN`. 

`INVALID_TREASURY_ACCOUNT_FOR_TOKEN` was thrown only if  `INVALID_ACCOUNT_ID` or `ACCOUNT_DELETED` is thrown.